content/livesync/quickstart.textile

```diff
@@ -1,7 +1,228 @@
+
+```[javascript]
+  const onUpdate = (err: Error | null, post?: PostType) => {
+    if (err) {
+      console.error(err);
+      return;
+    }
+
+    setPost(post!);
+  };
+```
+
+h3(#backend). The backend
+
+There are two parts of the backend that are key for the LiveSync integration in the livecomments demo. These two are the @sync@ function and the @outbox entry@.
+
+h4(#sync-function). Sync function
+
+The sync function in the backend is an endpoint of a @GET@ request, which returns the "@sequenceId@":/livesync/models#sequence-id, used to identify the point in the stream of change events that corresponds to the current version of the database's state. This "endpoint":https://github.com/ably-labs/models/blob/main/examples/livecomments/app/api/posts/%5Bid%5D/route.ts#L5-L18 also returns the current state of the data relevant to the expected data model, in this instance it's the current state of the post defined by its @id@.
+
+
+```[javascript]
+  export async function GET(request: NextRequest, { params }: { params: { id: string } }) {
+    try {
+      let id: number;
+      try {
+        id = Number(params.id);
+      } catch (error) {
+        return NextResponse.json({ message: 'failed to read :id url parameter', error }, { status: 400 });
+      }
+      const [data, sequenceID] = await getPost(id);
+      return NextResponse.json({ sequenceID, data });
+    } catch (error) {
+      return NextResponse.json({ message: 'failed to get post', error }, { status: 500 });
+    }
+  }
+```
+
+h4(#outbox-entry). outbox entry.
+
+The outbox entry is an entry of an update to one of the models within the database. In "this example":https://github.com/ably-labs/models/blob/main/examples/livecomments/lib/prisma/api.ts#L129-L139 it's a record of a comment being added to, updated with, or deleted from a post. When a post is added, updated, or deleted a record of this is also added to the outbox table so that the ADBC can pick this up and publish the update to the specified Pub/Sub channel.
+
+```[javascript]
+  export async function withOutboxWrite(
+    op: (tx: TxClient, ...args: any[]) => Promise<Prisma.outboxCreateInput>,
+    ...args: any[]
+  ) {
+    return await prisma.$transaction(async (tx) => {
+      const { mutation_id, channel, name, data, headers } = await op(tx, ...args);
+      await tx.outbox.create({
+        data: { mutation_id, channel, name, data, headers },
+      });
+    });
+  }
+```
+
+h2(#quickstart). Quickstart
+
+In this quickstart you will cover all the steps required to get an existing demo running locally on your machine.
+
+h3(#step-0). The project
+
+Clone the models repository (https://github.com/ably-labs/models/) and navigate to the @livecomments@ directory within @examples@. The @livecomments@ directory houses the project used for this working demo.
```

Comment:
I think a href directly to close should be good.

Reply:
I'm wondering if it's worth cloning the entire models repo just to be able to use the quickstart guide. Are there other options here?

Reply:
The only other option for this demo is to pull the example out into its own repository and then link to that/clone from that.